cmd/operator-sdk/new/cmd.go

```diff
@@ -129,7 +128,25 @@ func newFunc(cmd *cobra.Command, args []string) error {
 
 	switch operatorType {
 	case projutil.OperatorTypeAnsible:
-		if err := doAnsibleScaffold(); err != nil {
+		err := os.MkdirAll(projectName, 0755)
+		if err != nil {
+			log.Fatal(err)
+		}
+		// go inside of the project dir
+		err = os.Chdir(filepath.Join(projutil.MustGetwd(), projectName))
+		if err != nil {
+			log.Fatal(err)
+		}
+		cfg := input.Config{
+			AbsProjectPath: filepath.Join(projutil.MustGetwd()),
+			ProjectName:    projectName,
+		}
+
+		if err != nil {
+			return fmt.Errorf("invalid apiVersion and kind: %v", err)
+		}
```

Comment:
We do not need to do this check here. It still made in the https://github.com/operator-framework/operator-sdk/blob/1c1386f02ea76918f10ec5d3632b0c195e91bd01/internal/flags/apiflags/flags.go#L91

When we add the helm plugin the validation also will be made there and then we can remove the APIFlags from the project. See: https://github.com/operator-framework/operator-sdk/blob/master/internal/plugins/helm/flags.go#L76

Reply:
Nice catch.